velox/functions/prestosql/aggregates/MapUnionSumAggregate.cpp

```diff
@@ -68,7 +72,38 @@ struct Accumulator {
       if constexpr (std::is_same_v<S, double> || std::is_same_v<S, float>) {
         sums[key] += value;
       } else {
-        sums[key] = functions::checkedPlus<S>(sums[key], value);
+        S checkedSum;
+        auto overflow = __builtin_add_overflow(sums[key], value, &checkedSum);
+
+        if (UNLIKELY(overflow)) {
+          auto errorValue = (int128_t(sums[key]) + int128_t(value));
+
+          switch (valueKind) {
+            case TypeKind::TINYINT:
+              VELOX_ARITHMETIC_ERROR(
+                  "Value {} exceeds {}",
```

Comment:
Can these be an underflow?

Reply:
@mbasmanova Yes - we can get here due to an underflow as well. Currently, we get the incorrect error message below. I will update the logic to check for this and print the correct statement.

>  'Value -32788 exceeds 32767'

Reply:
I've updated the error-handling logic and added underflow tests for each type.

Reply:
@rrando901 Curious if Presto has this problem as well? What are some examples of error messages from Presto for overflow and underflow?

Reply:
@mbasmanova Looks like Presto handles underflow. Here's an example from `TinyintType.java`
```cpp
    public void writeLong(BlockBuilder blockBuilder, long value)
    {
        if (value > Byte.MAX_VALUE) {
            throw new GenericInternalException(format("Value %d exceeds MAX_BYTE", value));
        }
        else if (value < Byte.MIN_VALUE) {
            throw new GenericInternalException(format("Value %d is less than MIN_BYTE", value));
        }

        blockBuilder.writeByte((int) value).closeEntry();
    }
```